mlir/include/mlir-c/Dialect/Rock.h

```diff
@@ -0,0 +1,64 @@
+//===-- mlir-c/Dialect/Rock.h - C API for Rock dialect --------*- C
+//-*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM
```

Comment:
Nit: part of the rocMLIR project, and add a `Copyright (c) 2022 Advanced Micro Devices Inc.` to the bottom of the header bit

Reply:
I can see that looks more like what the standard APACHE license template says, but different from the other part of MLIR project. Can you let me know what's the right policy for the license header?

Reply:
That's a @mvermeulen question, really, and we're probably due for a mechanical PR that fixes up our header comments

I've taken to using "rocMLIR project" and putting an AMD copyright in new files